### PR TITLE
A lot of new classes

### PR DESCRIPTION
**Pull request by pyrox0**
Pulls a lot of datafixer classes in. I also took the chance to look through all of n/m/unmapped and map any class names that I could. This drops the number of classes in `unmapped` from 367 to 287, according to enigma's mapping stats. There's also a few other mappings I thought would be good to have in here, the main part being mapping ~92%~ of `n/m/util/profiling`.

### CODE REVIEW

**Comment by eristhea** [comment]
great work, we love to see unmapped classes going down! will review sometime today

File: mappings/net/minecraft/client/gui/screen/Bans.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/screen/Bans.mapping:1]: This doesn't contain bans
mojmap is `BanNoticeScreens`
I think `ConfirmBanScreenFactories` would be better

File: mappings/net/minecraft/client/render/entity/BoggedEntityRenderer.mapping [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/unmapped/C_ddcesbvk net/minecraft/client/render/entity/BoggedEntityRenderer
+	FIELD f_bcleisgn BOGGED_OVERLAY_TEXTURE Lnet/minecraft/unmapped/C_ncpywfca;
+	FIELD f_dsudahle BOGGED_TEXTURE Lnet/minecraft/unmapped/C_ncpywfca;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/render/entity/BoggedEntityRenderer.mapping]:
```suggestion
	FIELD f_bcleisgn OVERLAY_TEXTURE Lnet/minecraft/unmapped/C_ncpywfca;
	FIELD f_dsudahle TEXTURE Lnet/minecraft/unmapped/C_ncpywfca;
```
less redundant and more consistent with the handful of other entity renderers I checked

File: mappings/net/minecraft/datafixer/FixUtil.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/datafixer/FixUtil.mapping:1]: Since this is used outside of the `fix` package, I think `DataFixerUtil` ~~(or even just `Util`)~~ would be better

File: mappings/net/minecraft/client/gui/screen/Bans.mapping [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CLASS net/minecraft/unmapped/C_tkoofxmf net/minecraft/client/gui/screen/Bans
```
Comment (IotaBread) [inline mappings/net/minecraft/client/gui/screen/Bans.mapping]: Needs a more descriptive name such as `BanScreens`

File: mappings/net/minecraft/datafixer/fix/ComponentFix.mapping [hunk 3]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/unmapped/C_dxmxprhk net/minecraft/datafixer/fix/ComponentFix
```
Comment (IotaBread) [inline mappings/net/minecraft/datafixer/fix/ComponentFix.mapping]: I feel like this name needs a bit more information

File: mappings/net/minecraft/entity/vehicle/VehicleEntity.mapping [hunk 4]
```diff
@@ -8,8 +8,8 @@ CLASS net/minecraft/unmapped/C_ogoqhclb net/minecraft/entity/vehicle/VehicleEnti
 	METHOD m_gbralwah isDestroyedBy (Lnet/minecraft/unmapped/C_sbxfkpyv;)Z
 		ARG 1 source
 	METHOD m_qobhgcrb killAndDropItem (Lnet/minecraft/unmapped/C_vorddnax;)V
-	METHOD m_tqtorwvl killAndDropSelf (Lnet/minecraft/unmapped/C_sbxfkpyv;)V
-		ARG 1 source
+	METHOD m_tqtorwvl onDamage (Lnet/minecraft/unmapped/C_sbxfkpyv;)V
```
Comment (IotaBread) [inline mappings/net/minecraft/entity/vehicle/VehicleEntity.mapping]: Considering the old name, is this method ran every time the vehicle is damaged?

File: mappings/net/minecraft/client/gui/screen/Bans.mapping (hunk 2 above)
Comment (pyrox0) [reply mappings/net/minecraft/client/gui/screen/Bans.mapping]: Updated to `BanScreen`.

File: mappings/net/minecraft/datafixer/fix/ComponentFix.mapping (hunk 3 above)
Comment (pyrox0) [reply mappings/net/minecraft/datafixer/fix/ComponentFix.mapping]: Updated this to match Mojang's name of `ItemStackComponentRemainderFix`.

File: mappings/net/minecraft/entity/vehicle/VehicleEntity.mapping (hunk 4 above)
Comment (pyrox0) [reply mappings/net/minecraft/entity/vehicle/VehicleEntity.mapping]: I accidentially mis-mapped this while mapping something else. Reverted to the old name as the new one wasn't correct.

File: mappings/net/minecraft/client/gui/screen/BanScreen.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/gui/screen/BanScreen.mapping:1]: I think `ConfirmBanScreenFactories` would be better, it doesn't `extend Screen`, it just has static factory methods

**Review by IotaBread — APPROVED** [review]

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]
Not necessarily something for this PR, but the suffixes for `LootFunction` sub-classes are a bit inconsistent.
Most have `LootFunction`, a few have `Function`.
I'd prefer no suffix for the non-abstract sub-classes because their verb-prefixed names already imply their function-ness.

File: mappings/net/minecraft/client/gui/screen/Bans.mapping
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/client/gui/screen/Bans.mapping:1]: Based on old commit

File: mappings/net/minecraft/datafixer/FixUtil.mapping
Comment (eristhea) [reply mappings/net/minecraft/datafixer/FixUtil.mapping:1]: not a fan of just naming it `Util`, since that opens up people getting the wrong one in IDE autocomplete

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/datafixer/FixUtil.mapping:1]: fair enough

File: mappings/net/minecraft/client/render/entity/BoggedEntityRenderer.mapping (hunk 1 above)
Comment (pyrox0) [reply mappings/net/minecraft/client/render/entity/BoggedEntityRenderer.mapping]: Fixed!

File: mappings/net/minecraft/datafixer/FixUtil.mapping
Comment (pyrox0) [reply mappings/net/minecraft/datafixer/FixUtil.mapping:1]: Good point about the name being ambiguous. Mojang calls it `ExtraDataFixUtils`, so how about `DataFixUtil`?

File: mappings/net/minecraft/client/gui/screen/BanScreen.mapping
Comment (pyrox0) [reply mappings/net/minecraft/client/gui/screen/BanScreen.mapping:1]: Updated this to `BanScreenFactories`, as it doesn't just deal with confirmation screens for bans, but getting durations and descriptions of them as well.

**Comment by pyrox0** [comment]
> Not necessarily something for this PR, but the suffixes for `LootFunction` sub-classes are a bit inconsistent. Most have `LootFunction`, a few have `Function`. I'd prefer no suffix for the non-abstract sub-classes because their verb-prefixed names already imply their function-ness.

I can do this in a later PR.

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/component/DataComponentTypes.mapping [hunk 5]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/unmapped/C_icwdaizi net/minecraft/component/DataComponentTypes
```
Comment (eristhea) [inline mappings/net/minecraft/component/DataComponentTypes.mapping]: this is going to create a conflict with my pr :p

File: mappings/net/minecraft/entity/vehicle/VehicleEntity.mapping [hunk 6]
```diff
@@ -8,8 +8,8 @@ CLASS net/minecraft/unmapped/C_ogoqhclb net/minecraft/entity/vehicle/VehicleEnti
 	METHOD m_gbralwah isDestroyedBy (Lnet/minecraft/unmapped/C_sbxfkpyv;)Z
 		ARG 1 source
 	METHOD m_qobhgcrb killAndDropItem (Lnet/minecraft/unmapped/C_vorddnax;)V
-	METHOD m_tqtorwvl killAndDropSelf (Lnet/minecraft/unmapped/C_sbxfkpyv;)V
-		ARG 1 source
+	METHOD m_tqtorwvl killAndDropItem (Lnet/minecraft/unmapped/C_sbxfkpyv;)V
```
Comment (eristhea) [inline mappings/net/minecraft/entity/vehicle/VehicleEntity.mapping]: why? I'm pretty sure I mapped this, it's not a bad name

File: mappings/net/minecraft/registry/Registry.mapping [hunk 7]
```diff
@@ -39,6 +39,8 @@ CLASS net/minecraft/unmapped/C_tqxyjqsk net/minecraft/registry/Registry
 		ARG 1 id
 		ARG 2 entry
 	METHOD m_nqoydeav getEntries ()Ljava/util/Set;
+	METHOD m_nxhthmir getId (Lnet/minecraft/unmapped/C_ncpywfca;)Ljava/util/Optional;
+		ARG 1 identifier
```
Comment (eristhea) [inline mappings/net/minecraft/registry/Registry.mapping]:
```suggestion
		ARG 1 id
```

File: mappings/net/minecraft/registry/Registry.mapping [hunk 8]
```diff
@@ -39,6 +39,8 @@ CLASS net/minecraft/unmapped/C_tqxyjqsk net/minecraft/registry/Registry
 		ARG 1 id
 		ARG 2 entry
 	METHOD m_nqoydeav getEntries ()Ljava/util/Set;
+	METHOD m_nxhthmir getId (Lnet/minecraft/unmapped/C_ncpywfca;)Ljava/util/Optional;
```
Comment (eristhea) [inline mappings/net/minecraft/registry/Registry.mapping]: this doesn't return an identifier, it returns a holder reference
checking the code, I'd simply call this `getHolder`
```suggestion
	METHOD m_nxhthmir getHolder (Lnet/minecraft/unmapped/C_ncpywfca;)Ljava/util/Optional;
```

**Review by eristhea — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/component/DataComponentTypes.mapping (hunk 5 above)
Comment (pyrox0) [reply mappings/net/minecraft/component/DataComponentTypes.mapping]: Removed it from mine!

**Comment by pyrox0** [comment]
All suggestions added!

File: mappings/net/minecraft/registry/Registerable.mapping [hunk 9]
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/unmapped/C_hqoyyfco net/minecraft/registry/Registerable
```
Comment (eristhea) [inline mappings/net/minecraft/registry/Registerable.mapping]: I'm pretty sure this class used to be BootstrapContext, why the change?

Comment (pyrox0) [reply mappings/net/minecraft/registry/Registerable.mapping]: I don't remember changing this, but I can't find any trace of either a `BootstrapContext` or `Registerable` class in the past few branches. I'll change this to BootstrapContext.

Comment (eristhea) [reply mappings/net/minecraft/registry/Registerable.mapping]: works for me! I'll look into it a little more when I have access to enigma

**Review by eristhea — APPROVED** [review]

**Comment by pyrox0** [comment]
Forgot to actually push the above mentioned change, just did so.

**Comment by eristhea** [comment]
just an aside: bootstrap context was indeed the old name. it's passed to static methods in worldgen classes to register their features and structures. I'd add an automap rule for `net/minecraft/unmapped/C_hqoyyfco` to `context` in `simple_type_field_names.json5`

**Comment by pyrox0** [comment]
Added that!
